pkg/util/kubernetes/autoscalers/datadogexternal_util.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package autoscalers
+
+import (
+	"strconv"
+	"sync"
+	"time"
+
+	le "github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/leaderelection/metrics"
+)
+
+type minRemainingRequests struct {
+	sync.Mutex
+	val            int
+	timestamp      time.Time
+	expiryDuration time.Duration
+}
+
+func newMinRemainingRequests(expiryDuration time.Duration) minRemainingRequests {
+	return minRemainingRequests{
+		val:            -1,
+		timestamp:      time.Now(),
+		expiryDuration: expiryDuration,
+	}
+}
+
+func (mrr *minRemainingRequests) update(newVal string) {
+	mrr.Lock()
+	defer mrr.Unlock()
+
+	newValFloat, err := strconv.Atoi(newVal)
+
+	if err != nil {
+		return
+	}
+
+	isSet := mrr.val >= 0
+	hasExpired := time.Since(mrr.timestamp) > mrr.expiryDuration
+
+	if mrr.val >= newValFloat || !isSet || hasExpired {
```

Comment:
For readability we usually put the comparison with the same logical intent. In our case we want to store if `newVal is lower than existing val`, which is expressed more clearly with `newVal < mrr.val`